forc-plugins/forc-client/src/op/deploy.rs

```diff
@@ -197,46 +202,43 @@ pub async fn deploy(command: cmd::Deploy) -> Result<Vec<DeployedContract>> {
     Ok(contract_ids)
 }
 
-/// Applies specified target information to the provided arguments.
-///
-/// Basically provides preset configurations for known test-nets.
-fn apply_target(command: cmd::Deploy) -> Result<cmd::Deploy> {
-    let deploy_to_latest_testnet = command.testnet;
-    let target = if deploy_to_latest_testnet {
-        if command.target.is_some() {
-            bail!("Both `--testnet` and `--target` were specified: must choose one")
-        }
-        Some(Target::Beta3)
+/// Returns the gas to use for deployment, overriding default values if necessary.
+fn get_gas(command: &cmd::Deploy, node_info: NodeInfo, chain_info: ChainInfo) -> Gas {
+    // TODO: write unit tests for this function once https://github.com/FuelLabs/fuel-core/issues/1312 is resolved.
+    let price = if command.gas.price == 0 {
+        node_info.min_gas_price
     } else {
-        command.target.clone()
+        command.gas.price
     };
 
-    if let Some(target) = target {
-        match target {
-            cmd::deploy::Target::Beta2 | cmd::deploy::Target::Beta3 => {
-                // If the user did not specified a gas price, we can use `1` as a gas price for
-                // beta test-nets.
-                let gas_price = if command.gas.price == 0 {
-                    1
-                } else {
-                    command.gas.price
-                };
-
-                let target_url = Some(target.target_url().to_string());
-                Ok(cmd::Deploy {
-                    gas: Gas {
-                        price: gas_price,
-                        ..command.gas
-                    },
-                    node_url: target_url,
-                    ..command
-                })
-            }
-            cmd::deploy::Target::LATEST => Ok(command),
-        }
+    let limit = if command.gas.limit == 0 {
```

Comment:
```suggestion
    let limit = if command.gas.limit == fuel_tx::ConsensusParameters::DEFAULT.max_gas_per_tx {
```

command.gas.limit is `fuel_tx::ConsensusParameters::DEFAULT.max_gas_per_tx` by default. So if the user uses `forc-deploy --testnet` currently that if is not checked and following error is returned:

```console
error: Decode error: Custom { kind: Other, error: "Response errors; TransactionGasLimit" }
```

I think we are good to go with only changing that 0 to `fuel_tx::ConsensusParameters::DEFAULT.max_gas_per_tx `, later on once we clean forc-tx stuff too this would be cleaned as well

Reply:
I'm making 0 the default so that we can check if it's 0 to override it. Otherwise, if `fuel_tx::ConsensusParameters::DEFAULT.max_gas_per_tx` is the default, and the user actually wants to use that value, we will not know whether the user provided it or not, so we will override incorrectly. In the previous code, it was impossible for the user to actually run it with `fuel_tx::ConsensusParameters::DEFAULT.max_gas_per_tx`

Reply:
👌 sounds good to me for getting things working, but ideally I feel like they should be optional without default values and if user did not provide a value for them we should get them from the node.

Here is where the gas limit default value is specified:
https://github.com/FuelLabs/sway/blob/149b65726298b008b612612368e2b3e0e9b5997c/forc-plugins/forc-tx/src/lib.rs#L108-L109

Reply:
Good catch, I forgot to set the default to 0. I think it's better to default to 0 so that users can still specify `fuel_tx::ConsensusParameters::DEFAULT.max_gas_per_tx ` if they want, and we won't override it.